ring_buffer_it_test.go

```diff
@@ -0,0 +1,187 @@
+		assert.NoError(t, err)
+		assert.Equal(t, int64(0), sequence)
+
+		actualItem, err := rb.ReadOne(context.Background(), 0)
+		assert.NoError(t, err)
+		assert.Equal(t, "foo", actualItem)
+	})
+}
+
+func TestRingbuffer_AddNilElement(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		_, err := rb.Add(context.Background(), nil, hz.OverflowPolicyFail)
+		assert.Error(t, err)
+	})
+}
+
+func TestRingbuffer_AddAll(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		_, err := rb.AddAll(context.Background(), hz.OverflowPolicyOverwrite, "foo", "bar")
+		assert.NoError(t, err)
+
+		foo, err := rb.ReadOne(context.Background(), 0)
+		assert.NoError(t, err)
+		assert.Equal(t, "foo", foo)
+
+		bar, err := rb.ReadOne(context.Background(), 1)
+		assert.NoError(t, err)
+		assert.Equal(t, "bar", bar)
+	})
+}
+
+func TestRingbuffer_Size(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		_, err := rb.Add(context.Background(), "one", hz.OverflowPolicyOverwrite)
+		assert.NoError(t, err)
+		size, err := rb.Size(context.Background())
+		assert.NoError(t, err)
+		assert.Equal(t, int64(1), size)
+	})
+}
+
+func TestRingbuffer_Capacity(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		capacity, err := rb.Capacity(context.Background())
+		assert.NoError(t, err)
+		assert.GreaterOrEqual(t, capacity, int64(9999), "There should be a capacity of at least 9999 items.")
+	})
+}
+
+func TestRingbuffer_RemainingCapacity(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		capacity, err := rb.RemainingCapacity(context.Background())
+		assert.NoError(t, err)
+		assert.GreaterOrEqual(t, capacity, int64(9999), "There should be a remaining capacity of at least 9999 items.")
+	})
+}
+
+func TestRingbuffer_HeadSequence_and_TailSequence(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		rb.AddAll(context.Background(), hz.OverflowPolicyOverwrite, "one", "two", "three")
+
+		head, err := rb.HeadSequence(context.Background())
+		assert.NoError(t, err)
+		tail, err := rb.TailSequence(context.Background())
+		assert.NoError(t, err)
+
+		assert.Equal(t, int64(2), tail-head)
+	})
+}
+
+func TestRingbuffer_ReadMany_ReadCount(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		rb.AddAll(context.Background(), hz.OverflowPolicyOverwrite, "0", "1", "2", "x")
+
+		rs, err := rb.ReadMany(context.Background(), 0, 3, 3, nil)
+		assert.NoError(t, err)
+
+		assert.Equal(t, int32(3), rs.ReadCount())
+	})
+}
+
+func TestRingbuffer_ReadMany_Get_with_startSequence(t *testing.T) {
+	it.RingbufferTester(t, func(t *testing.T, rb *hz.Ringbuffer) {
+		rb.AddAll(context.Background(), hz.OverflowPolicyOverwrite, "x", "0", "1", "2", "y", "z")
+
+		rs, err := rb.ReadMany(context.Background(), 1, 3, 3, nil)
+		assert.NoError(t, err)
+
+		item, _ := rs.Get(0)
```

Comment:
It's better not to ignore the errors. You can use `it.MustValue`, like:
```
item := it.MustValue(rs.Get(0))
```
Similar comment to test code where the error is ignored.

Reply:
Done.
4130060823bf58a1344681792dc0f5d38e424d33